### PR TITLE
Default to most recent charm on bundle deploy

### PR DESCRIPTION
**Pull request by makyo**
Driveby: `make devel` now uses demo.jujucharms.com for bundle parsing.

### CODE REVIEW

**Comment by makyo** [comment]
# QA

1. run `make devel` and deploy a bundle, you should receive an uncommitted bundle.
2. deploy and expose the GUI in a live env
3. `juju set juju-gui juju-gui-source="https://github.com/makyo/juju-gui.git config-problem"`
4. search for "landscape dense maas" and deploy the bundle
5. Uncommitted state should work fine. At this point, you can delete the machine where the services are colocated so that they don't actually create units.
6. Click deploy.  Everything should work correctly. The two landscape charms should get the most recent version.

File: app/utils/bundle-importer.js [hunk 1]
```diff
@@ -353,7 +353,7 @@ YUI.add('bundle-importer', function(Y) {
           ghostService.set('config', config);
 
           this.env.deploy(
-              record.args[0],
+              charm.get('id'),
```
Comment (hatched) [inline app/utils/bundle-importer.js]: Plz add a comment here as to why this is required

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2622/
Test PASSed.

**Comment by hatched** [comment]
Thanks for this branch!
+1 with that comment request.
QAOK

QA Notes
Tested deploying bundle with devel and it worked as expected, changed configuration options pre-deploy and they persisted after deploy to the sandbox.

Followed the rest of the instructions for the landscape bundle

File: app/utils/bundle-importer.js (hunk 1 above)
Comment (mitechie) [inline app/utils/bundle-importer.js]: +1 this is the meat and potatoes of this fix so good to comment on why it's strange.

**Comment by mitechie** [comment]
:+1: ty for the update

**Comment by makyo** [comment]
:shipit: Thanks!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2623/
Test PASSed.

**Comment by makyo** [comment]
Fixes https://bugs.launchpad.net/juju-gui/+bug/1463502

**Pull request merged**